api/envoy/api/v2/listener/listener.proto

```diff
@@ -216,4 +227,9 @@ message ListenerFilter {
 
     google.protobuf.Any typed_config = 3;
   }
+
+  // Decide when to enable/disable this listener filter on incoming traffic.
+  oneof switches {
```

Comment:
Why is this `oneof`?

Reply:
I have the concern one day user prefer whitelist to the this blacklist. Is it a over concern?

Reply:
Instead of a `oneof` could we have logical operators on the matchers themselves? If the worry is changing the nature of the match it seems odd to base that on the field rather than an operator.

Also what if you'd want some whitelist matchers and some blacklist matchers on the same filter?

Reply:
The `disabled` field is to make easy things easy. It's the only requirement as for now.

The idea of `oneof` is to make hard things possible and semantics is clear : all the `whitelist` and `blacklist` should locate and at most one.
I like logical operator. It could be the future member inside switches. 
oneof {
   disabled, // blacklist
   enabled,  // whitelist
   operator,  // a complex Boolean algebra if needed
}
However, I think it's over design if I introduce a bool engine at this moment :)

Reply:
I think the API design theme is around economy of mechanism; we don't want multiple ways of doing the same thing. It sounds like here that in the interest of not putting too much upfront work into the API we're setting ourselves up for a messier future. I would prefer we did this more systematically, similar to how other matchers work in the API, as suggested by @junr03.

Reply:
Below should cover all the combination in the system.
switches : {
 // the matchers will be evaluated using OR operator
 // All the fields in each matcher will be evaluated using AND operator.
 // Currently we only have A = destination_port so B is always true.
 // (matchers[0].A AND matchers[0].B) OR (matchers[1].A and matchers[1].B) OR ...
 repeated ListenerFilterMatch matchers; 
}
@htuch @junr03